core/roboconf-core/src/main/java/net/roboconf/core/utils/Utils.java

```diff
@@ -120,33 +120,45 @@ public static boolean areEqual( Object o1, Object o2 ) {
 		return result;
 	}
 
+
 	/**
 	 * Expand a template, replacing each {{ param }} by the corresponding value.
+	 * <p>
 	 * Eg. "My name is {{ name }}" will result in "My name is Bond", provided that "params" contains "name=Bond".
-	 * @param s The template to expand
-	 * @param params The parameters to be expanded in the template
-	 * @return The expanded template.
+	 * </p>
+	 *
+	 * @param s the template to expand
+	 * @param params the parameters to be expanded in the template
+	 * @return the expanded template
 	 */
 	public static String expandTemplate(String s, Properties params) {
-		if(params == null || params.size() < 1) return s;
 
-		Pattern pattern = Pattern.compile( "\\{\\{\\s*\\S+\\s*\\}\\}" );
-		Matcher m = pattern.matcher(s);
+		String result;
+		if( params == null || params.size() < 1 ) {
+			result = s;
 
-		StringBuffer sb = new StringBuffer();
```

Comment:
The difference is that there is only one return statement in the method.
This is a good habit when you debug code and methods. By having a single return statement, you can put a single break point just before the return. This way, you know very quickly if it is this method that returns something wrong or not. Said differently, it saves time when you search where your code is not behaving correctly.

For code coverage, I simply completed unit tests (we wered not testing null and empty params).